sql/core/src/main/scala/org/apache/spark/sql/execution/HiveResult.scala

```diff
@@ -124,4 +124,46 @@ object HiveResult {
     case (other, _ : UserDefinedType[_]) => other.toString
     case (other, tpe) if primitiveTypes.contains(tpe) => other.toString
   }
+
+  /** Escape a String in JSON format. */
+  private def escapeString(str: String): String = {
```

Comment:
did you take this method from somewhere or is this created by you? If the first, can you say where this comes from? If not, can you check what Hive does? I assume they have similar code for this, so checking theirs seems safer to me.

Reply:
> you check what Hive does? I assume they have similar code for this, so checking theirs seems safer

It was taken from Hive. I'll add a comment there.